### PR TITLE
(Plugin-388) Fail Parquet output format validation if schema definition is null or empty.

### PR DESCRIPTION
**Pull request by supriyapatil-google**

### CODE REVIEW

**Review by CuriousVini — COMMENTED** [review]
Now that I took a closer look at the issue, it happens in CDF environment but not in sandbox. The reason being in CDF environment UI is sending schema as empty string.
I think what we need in the fix is to check if schema is an empty or null string then throw an error. Does this code change fixes issue in cdf environment?

image

image

File: format-parquet/pom.xml [hunk 1]
```diff
@@ -79,6 +79,10 @@
       <artifactId>jackson-mapper-asl</artifactId>
       <version>1.9.13</version>
     </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
```
Comment (CuriousVini) [inline format-parquet/pom.xml:83]: Dont think there is need to add this dependency explicitly.

**Review by CuriousVini — APPROVED** [review]
one minor comment. LGTM 👍

**Comment by supriyapatil-google** [comment]
Sample validation works in snapshot environment with my changes.

File: format-parquet/pom.xml (hunk 1 above)
Comment (supriyapatil-google) [reply format-parquet/pom.xml:83]: I do see this dependency explicitly specified in other plugin packages like (database-plugins, http-plugins, core-plugins) so its best to follow the same pattern.

Comment (CuriousVini) [reply format-parquet/pom.xml:83]: sounds good

**Review by CuriousVini — APPROVED** [review]
lgtm 👍  Please squash all the commits into one before merging.

**Comment by supriyapatil-google** [comment]
1. Added unit tests in FileSinkTestRun to validate empty and null value output schema values.
2. Overridden ParquetOutputFormatProvider#validate(FormatContext) to validate schemas as part of plugin validation.
3. Verified parquet plugin validation in Sandbox CDAP environment.

Please take another look. Thanks!

**Review by CuriousVini — APPROVED** [review]
lgtm 👍

**Pull request merged**